lambdas/handlers/create_document_reference_handler.py

```diff
@@ -58,6 +59,10 @@ def processing_event_details(event):
     try:
         body = json.loads(event["body"])
         nhs_number = body["subject"]["identifier"]["value"]
+        try:
+            validate_id(nhs_number)
+        except InvalidResourceIdException:
+            raise CreateDocumentRefException(400, "Invalid NHS number")
```

Comment:
`validate_id` is already used in the create reference service, as it is part of the business logic.